python-package/lightgbm/basic.py

```diff
@@ -41,20 +41,26 @@ def info(self, msg: str) -> None:
     def warning(self, msg: str) -> None:
         warnings.warn(msg, stacklevel=3)
 
+    def error(self, msg: str) -> None:
+        warnings.warn(msg, stacklevel=3)
+
 
-_LOGGER: Union[_DummyLogger, Logger] = _DummyLogger()
+_LOGGER: Any = _DummyLogger()
 
 
-def register_logger(logger: Logger) -> None:
+def register_logger(logger: Any) -> None:
```

Comment:
I think we can make this API even more flexible by introducing three new arguments with default values: `info_method_name`, `warning_method_name` and `error_method_name`. This will allow, for example, use `warning` or `warn` method depending on logger backend
> Note There is an obsolete method `warn` which is functionally identical to `warning`. As `warn` is deprecated, please do not use it - use `warning` instead.
   https://docs.python.org/3/library/logging.html#logging.Logger.warning

Reply:
I agree, will add these args.